wallet/txauthor/author.go

```diff
@@ -309,6 +323,54 @@ func spendWitnessKeyHash(txIn *wire.TxIn, pkScript []byte,
 	return nil
 }
 
+// spendTaprootKey generates, and sets a valid witness for spending the passed
+// pkScript with the specified input amount. The input amount *must*
+// correspond to the output value of the previous pkScript, or else verification
+// will fail since the new sighash digest algorithm defined in BIP0341 includes
+// the input value in the sighash.
+func spendTaprootKey(txIn *wire.TxIn, pkScript []byte,
+	inputValue int64, chainParams *chaincfg.Params, secrets SecretsSource,
+	tx *wire.MsgTx, hashCache *txscript.TxSigHashes, idx int) error {
+
+	// First obtain the key pair associated with this p2tr address.
+	_, addrs, _, err := txscript.ExtractPkScriptAddrs(pkScript, chainParams)
+	if err != nil {
+		return err
+	}
+	privKey, _, err := secrets.GetKey(addrs[0])
+	if err != nil {
+		return err
+	}
+	pubKey := privKey.PubKey()
+
+	tapKey := txscript.ComputeTaprootKeyNoScript(pubKey)
+	p2trAddr, err := btcutil.NewAddressTaproot(
+		schnorr.SerializePubKey(tapKey), chainParams,
+	)
+	if err != nil {
+		return err
+	}
+
+	// With the concrete address type, we can now generate the
```

Comment:
FWIW, we could save a few lines here just by using the passed `pkScript` here below when calling `TaprootWitnessSignature`.